frame.js

```diff
@@ -0,0 +1,74 @@
+class Frame {
+  constructor() {
+    this.rollCounter = 1
+    this.rolls = []
+    this.score = 0
+    this.live = true
+  }
+
+  getRollCounter() {
+    return this.rollCounter;
+  }
+  getScore() {
+    return this.score;
+  }
+  isLive() {
+    return this.live;
+  }
+  addRoll() {
+    return this.rollCounter++;
+  }
+  addPinsToScore(pins) {
+    return this.score += pins;
+  }
+  minusRoll() {
+    return this.rollCounter--;
+  }
+  getRolls() {
+    return this.rolls;
+  }
+  addPinsToRolls(pins) {
+  return this.rolls.push(pins);
+  }
+  isStrike() {
+    return this.rolls[0] === 10;
+  }
+  isSpare() {
+    return this.rolls.reduce((roll_one, roll_two) => roll_one + roll_two) === 10 && !this.isStrike();
```

Comment:
This is a very complicated way of writing `this.rolls[0] + this.rolls[1] === 10`